### PR TITLE
chore: update controller tools from v0.11.0 to v0.14.0

### PR DESCRIPTION
**Pull request by basti1302**
This should fix the build error that currently happens: panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa[40]

### CODE REVIEW

File: config/samples/kustomization.yaml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 ## Append samples you want in your CSV to this file as resources ##
 resources:
-  - v1alpha1_perses.yaml
+  - perses.dev_v1alpha1_perses.yaml
+  - perses.dev_v1alpha1_persesdashboard.yaml
+  - perses.dev_v1alpha1_persesdashboard-2.yaml
```
Comment (Nexucis) [inline config/samples/kustomization.yaml:5]: `perses.dev_v1alpha1_persesdashboard-2.yaml` shouldn't be removed ?

Comment (basti1302) [reply config/samples/kustomization.yaml:5]: Well it is a file that exists in config/samples. The only file listed here before `v1alpha1_perses.yaml`, which does no longer exist (was removed in commit #45e3c38800a4200f00f74badbf8aebf28867aa82).

I can remove the reference to `perses.dev_v1alpha1_persesdashboard-2.yaml` if you want. Should the file itself also be removed? It has been added in #45e3c38800a4200f00f74badbf8aebf28867aa82 as well, by @jgbernalp  (It seems it is mostly a copy of `perses.dev_v1alpha1_persesdashboard.yaml`?)

Comment (Nexucis) [reply config/samples/kustomization.yaml:5]: looks weird to me yeah.
Sorry if you find my question dumb, I'm not that familiar with the operator. I would wait what @jgbernalp will say about it

Comment (basti1302) [reply config/samples/kustomization.yaml:5]: No worries. I don't have any opinion on these samples either. I merely saw that the kustomization.yaml was outdated and referenced things that no longer exist, so I updated it with everything that exists in the directory 🤷

Comment (jgbernalp) [reply config/samples/kustomization.yaml:5]: We can remove it is the same as the first example so does not add any value at the moment.

Comment (basti1302) [reply config/samples/kustomization.yaml]: Removed both the file and the reference.

**Review by Nexucis — APPROVED** [review]
awesome, thank you @basti1302 !

**Comment by Nexucis** [comment]
for the linter issue, I think it's more related to golanglint-ci itself. WIll merge and adress the issue when I can

**Comment by basti1302** [comment]
Thanks for merging. 

> for the linter issue, I think it's more related to golanglint-ci itself. WIll merge and adress the issue when I can

Right. I _think_ the linting fails because it is executed with Go 1.23, and it might work with an older Go version (1.22 or 1.21). I had similar issues in another project of mine as well; it seems golanglint-ci is picky about the Go version it is executed with. But I am not sure where the Go version for the linting step is set, it seems to come from either `- uses: perses/github-actions@v0.1.0` or `- uses: ./.github/perses-ci/actions/setup_environment`. 🤷

**Pull request merged**